tests/lib/azure/fixture/fixture.go

```diff
@@ -50,31 +55,41 @@ func Run(
 	location string,
 	testfunc Test,
 ) {
+	//FIXME: We could remove testname on Go 1.8
 	t.Run(testname, func(t *testing.T) {
 		t.Parallel()
 		ctx, cancel := context.WithTimeout(context.Background(), timeout)
 		defer cancel()
 
+		logger := log.New(t, testname)
+		defer logger.Close()
+
 		session := azure.NewSession(t)
 		resgroup := fmt.Sprintf("klb-test-fixture-%s-%d", testname, rand.Intn(9999999))
 
-		resources := azure.NewResourceGroup(ctx, t, session)
+		resources := azure.NewResourceGroup(ctx, t, session, logger)
 		defer func() {
 			// We cant use an expired context when cleaning up state from Azure.
-			ctx, cancel := context.WithTimeout(context.Background(), timeout)
+			ctx, cancel := context.WithTimeout(context.Background(), timeout/2)
```

Comment:
If you configure a timeout for the test for 5 minutes it would try to cleanup for 5 minutes too, it was taking too long :-). It could be another parameter...but we already have so much parameters x_x